storage/innobase/buf/buf0flu.cc

```diff
@@ -2611,25 +2611,24 @@ static
 void
 page_cleaner_adapt_lru_sleep_time(
 /*==============================*/
-	ulint*	lru_sleep_time,	/*!< in/out: desired page cleaner thread sleep
+	ulint*	lru_sleep_time)	/*!< in/out: desired page cleaner thread sleep
 				time for LRU flushes  */
-	ulint	lru_n_flushed) /*!< in: number of flushed in previous batch */
 
 {
 	ulint free_len = buf_get_total_free_list_length();
-	ulint max_free_len = srv_LRU_scan_depth * srv_buf_pool_instances;
+	ulint lru_len = UT_LIST_GET_LEN(buf_pool_from_array(0)->LRU);
```

Comment:
Almost the same expression used inside buf_flush_LRU_tail(void)  
ut_min(srv_LRU_scan_depth, UT_LIST_GET_LEN(buf_pool->LRU)); 
UT_LIST_GET_LEN(buf_pool->LRU) is not a constant and could be changed in runtime for 5.6.
Should I make a function similar to buf_get_total_free_list_length, e.g. buf_get_total_LRU_list_length ? Length parameter is cached in list and this check should be fast.